test-app/tests/integration/components/multiselect-test.gts

```diff
@@ -0,0 +1,1215 @@
+        data-multiselect
+        as |multiselect|
+      >
+        <multiselect.Option>{{multiselect.option}}</multiselect.Option>
+      </Multiselect>
+    </template>);
+
+    await click('[data-multiselect]');
+
+    assert.dom('[role="option"]').exists({ count: 2 });
+
+    await triggerKeyEvent('[data-multiselect]', 'keydown', 'ArrowDown');
+
+    assert
+      .dom('[role="option"]:first-child')
+      .hasAttribute('data-active', 'false');
+    assert
+      .dom('[role="option"]:last-child')
+      .hasAttribute('data-active', 'true');
+  });
+
+  test('it sets the "active" item to the previous item in the list when using the UP arrow', async function (assert) {
+    let selected = ['red'];
+
+    // NOTE: Setting the selected option to "red" here so that
+    // the last item in the list will be active so that we can
+    // move up!
+    await render(<template>
+      <Multiselect
+        @selected={{selected}}
+        @options={{testColors}}
+        @removeButtonLabelFunction={{removeButtonLabelFunction}}
+        data-multiselect
+        as |multiselect|
+      >
+        <multiselect.Option>{{multiselect.option}}</multiselect.Option>
+      </Multiselect>
+    </template>);
+
+    await click('[data-multiselect]');
+
+    assert.dom('[role="option"]').exists({ count: 2 });
+
+    await triggerKeyEvent('[data-multiselect]', 'keydown', 'ArrowUp');
+
+    assert
+      .dom('[role="option"]:first-child')
+      .hasAttribute('data-active', 'true');
+    assert
+      .dom('[role="option"]:last-child')
+      .hasAttribute('data-active', 'false');
+  });
+
+  test('it closes an open popover when the ESCAPE key is pressed', async function (assert) {
+    await render(<template>
+      <Multiselect
+        @options={{testColors}}
+        @removeButtonLabelFunction={{removeButtonLabelFunction}}
+        data-multiselect
+        as |multiselect|
+      >
+        <multiselect.Option>{{multiselect.option}}</multiselect.Option>
+      </Multiselect>
+    </template>);
+
+    await click('[data-multiselect]');
+
+    assert.dom('[role="listbox"]').exists();
+
+    await triggerKeyEvent('[data-multiselect]', 'keydown', 'Escape');
+
+    assert.dom('[role="listbox"]').doesNotExist();
+  });
+
+  test('it closes an open popover when the component is blurred', async function (assert) {
+    await render(<template>
+      {{! template-lint-disable require-input-label }}
+      <input placeholder="test" data-input />
+
+      <Multiselect
+        @options={{testColors}}
+        @removeButtonLabelFunction={{removeButtonLabelFunction}}
+        data-multiselect
+        as |multiselect|
+      >
+        <multiselect.Option>{{multiselect.option}}</multiselect.Option>
+      </Multiselect>
+    </template>);
+
+    // Open the popover
+    await click('[data-multiselect]');
+
+    assert.dom('[role="listbox"]').exists();
+
+    // Now blur the elment
+    await click('[data-input]');
```

Comment:
I'm confused, doesn't this focus the `<input />` element?

Reply:
Yeah I can add a comment here, but we are blurring the input on https://github.com/CrowdStrike/ember-toucan-core/pull/219/files/16695506e740b9b0240a57b5faf3f3f14193e104#diff-136cfdc764a84882ddf48cd853edc7085a558470ed9eb3d8b4d33de1adfa8795R958 (the non-multiselect-input), not the multiselect input!